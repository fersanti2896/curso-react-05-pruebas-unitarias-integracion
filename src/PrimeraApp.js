/* Functionals Components */
import React from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ( {saludo, subtitulo} ) => {
    // const saludo2 = { name: 'Fernado', age: 24};

    return (
        <>
            <h1> { saludo } </h1>
            {/* <pre> {JSON.stringify(saludo, null, 3)} </pre> */}
            <p> { subtitulo }</p>
        </>
    );
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    subtitulo: 'Soy un subtitulo'
}

export default PrimeraApp;