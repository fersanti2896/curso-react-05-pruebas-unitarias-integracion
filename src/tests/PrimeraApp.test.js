import React from 'react';
import { shallow } from 'enzyme'
import '@testing-library/jest-dom'
import PrimeraApp from '../PrimeraApp'


describe('Pruebas en Componenten PrimeraApp', () => {
    /* Prueba con jest-dom */
    // test('Debe mostrar "Hola, soy Fernando"', () => {
    //     const saludo = 'Hola, soy Fernando';
    //     const { getByText } = render(<PrimeraApp saludo = {saludo}/>);

    //     expect(getByText(saludo)).toBeInTheDocument();
    // });

    /* Pruebas con Enzyme*/
    test('Debe mostrar <PrimeraApp /> correctamente', () => {
        const saludo = 'Hola, soy Fernando';
        const wrapper = shallow(<PrimeraApp saludo = {saludo}/>);

        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar el subtitulo enviado por props', () => {
        const saludo = 'Hola, soy Fernando';
        const subTitulo = 'Soy el curso de React'
        const wrapper = shallow(
            <PrimeraApp saludo = {saludo} subtitulo = {subTitulo}/>
        );

        const textoParrafo = wrapper.find('p').text();
        
        expect(textoParrafo).toBe(subTitulo);
    })
    
        
});