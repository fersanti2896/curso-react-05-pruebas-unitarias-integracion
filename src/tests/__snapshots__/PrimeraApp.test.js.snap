// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Componenten PrimeraApp Debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, soy Fernando
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
